### ISSUE TITLE
[Bug Report]: Deploying policyassignment on resourcegroup using a root bicep module fails

### ISSUE THREAD

**Issue by tkwaks** [Azure/ResourceModules#2493]
### Describe the bug

When deploying a root bicep file to targetScope = 'managementGroup' Or targetScope = 'subscription' all is well.

When we try to deploy an Authorization PolicyAssignments ResourceGroup using a root module at a resourcegroup level we recieve an error message: 

New-AzResourceGroupDeployment : 12:05:16 PM - Error: Code=InvalidTemplate; Message=Deployment template validation failed: 'The template resource 'wfwdqi76fhxhs-PolicyAssignment-MG-Module' at line 
'161' and column '9' is not valid: The language expression property 'location' doesn't exist, available properties are 'name, properties'.. Please see https://aka.ms/arm-template-expressions for usage 
details.'.
At line:2 char:1
+ New-AzResourceGroupDeployment -TemplateFile 'C:\dev\azure-resources\m ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-AzResourceGroupDeployment], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDeploymentCmdlet
 
New-AzResourceGroupDeployment : The deployment validation failed
At line:2 char:1
+ New-AzResourceGroupDeployment -TemplateFile 'C:\dev\azure-resources\m ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzResourceGroupDeployment], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDeploymentCmdlet
 
We See two issues":

1)
- when building the root bicep file a json file is created with a "location": "[deployment().location]" this deployment().location does not seem te be working with the New-AzResourceGroupDeployment
- when building the resource bicep file a json file is created with a "location": "[parameters('location')]" it works with New-AzResourceGroupDeployment

2) 
- when building the root bicep file the resource type": "Microsoft.Resources/deployments" contains a "location": "[parameters('location')]", this gives a error location is not allowed
- when build the resource bicep file the resource type": "Microsoft.Resources/deployments" does not contain a "location" parameter.

### To reproduce

[policyAssignments.zip](https://github.com/Azure/ResourceModules/files/10287653/policyAssignments.zip)

New-AzManagementGroupDeployment -TemplateFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\deploy.bicep' -TemplateParameterFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\.test\mgmic.parameters.json' -Location 'westeurope' -ManagementGroupId 'ManagementGroupName' 

New-AzSubscriptionDeployment -TemplateFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\deploy.bicep' -TemplateParameterFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\.test\submic.parameters.json' -Location 'westeurope'

New-AzResourceGroupDeployment -TemplateFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\deploy.bicep' -TemplateParameterFile 'C:\dev\azure-resources\modules\Microsoft.Authorization\policyAssignments\.test\rgmic.parameters.json' -ResourceGroupName 'ResourceGroupName'

**Comment by ahmadabdalla** [comment]
@tkwaks I've had a look at this issue,

I would like to confirm: Are you attempting to perform a 'Resource Group Scope' deployment using the 'overarching' or [policy assignment orchestrating module](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Authorization/policyAssignments/deploy.bicep). Is that correct?

I created the following parameter file:

```json

{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "name": {
            "value": "<<namePrefix>>${serviceShort}001"
        },
        "policyDefinitionId": {
            "value": "/providers/Microsoft.Authorization/policyDefinitions/06a78e20-9358-41c9-923c-fb736d382a4d"
        },
        "subscriptionId": {
            "value": "<<subscriptionId>>"
        },
        "resourceGroupName": {
            "value": "<<resourceGroupName>>"
        }
    }
}

```

I then performed a management group deployment using (New-AzManagementGroupDeployment). And it was successful. 

The [policy assignment orchestrating module](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Authorization/policyAssignments/deploy.bicep) is a 'management group scope' deployment and cannot be used with (New-AzResourceGroupDeployment).

If you would like to perform direct Resource Group Deployment, I would suggest you use the [Resource Group](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Authorization/policyAssignments/resourceGroup/deploy.bicep) one instead.
